src/razor/src/blazorDebug/blazorDebugConfigurationProvider.ts

```diff
@@ -179,6 +262,220 @@ export class BlazorDebugConfigurationProvider implements vscode.DebugConfigurati
+                    );
+                    const oldPid = BlazorDebugConfigurationProvider.pidsByUrl.get(urlStr);
+                    if (oldPid != undefined) {
+                        process.kill(oldPid);
+                    }
+                    BlazorDebugConfigurationProvider.pidsByUrl.set(urlStr, spawnedProxy.pid);
+                    const url = new URL(proxyUrlString);
+                    return [
+                        `${url.protocol.replace(`http`, `ws`)}//${url.host}{browserInspectUriPath}`,
+                        proxyICorDebugPort,
+                        proxyBrowserPort,
+                    ];
+                }
+            }
+
+            for await (const error of spawnedProxy.stderr) {
+                BlazorDebugConfigurationProvider.vsWebAssemblyBridgeOutputChannel.appendLine(`ERROR: ${error}`);
+            }
+        } catch (error: any) {
+            if (spawnedProxy.pid) {
+                BlazorDebugConfigurationProvider.vsWebAssemblyBridgeOutputChannel.appendLine(
+                    `Error occured while spawning debug proxy. Terminating debug proxy server.`
+                );
+                process.kill(spawnedProxy.pid);
+            }
+            throw error;
+        }
+        return ['', -1, -1];
+    }
+
+    private static async isNet9OrNewer(): Promise<boolean> {
+        let isNet9OrNewer = false;
+        const configurationsObject = {
+            $properties: ['BuildProperties'],
+            BuildProperties: {
+                $properties: ['Id', 'Name', 'Value', 'StorageType'],
+                $where: {
+                    '&&': [
+                        { '==': [{ '.': 'Name' }, 'TargetFramework'] },
+                        { '==': [{ '.': 'StorageType' }, 'ProjectFile'] },
+                    ],
+                },
+            },
+        };
+        const query = {
+            context: 'Projects',
+            query: {
+                $properties: ['ActiveConfigurations'],
+                ActiveConfigurations: configurationsObject,
+                $filter: { ProjectsByCapabilities: ['WebAssembly'] },
+            },
+        };
+
+        const queryString = JSON.stringify(query);
+        const proxy = await getCSharpDevKit()?.exports.serviceBroker.getProxy<IQueryExecutionService>(
+            Descriptors.projectQueryExecutionService
+        );
+        try {
+            if (proxy) {
+                const result = await proxy.ExecuteQueryAsync(queryString);
+                const queryResult = JSON.parse(result) as QueryResult<Project>;
+                if (queryResult && queryResult.Items) {
+                    isNet9OrNewer = true;
+                    queryResult.Items.forEach((project) => {
+                        project.ActiveConfigurations?.forEach((activeConfig) => {
+                            if (
+                                activeConfig.BuildProperties[0].Value == 'net5.0' ||
```

Comment:
Instead of a list of configurations which aren't .NET 9, shouldn't we use a regex to test if this is a `net<ver>` configuration (with anything other then a 'net<ver>' configuration saying "no"), and extract the version number, then use semver (or some simple hand rolled code to decide if that is '9.0' or newer.

Reply:
done.